### DISCUSSION TITLE
Converting the output of `cf.PlotVariables1D` as `ROOT` file

### DISCUSSION THREAD

**Discussion by gsaha009** [columnflow/columnflow#457] [General]
Hello CF Experts,

We are trying to estimate some Fake Factors in a columnflow based framework. We defined orthogonal categories and we saved several histograms in `pickle` files as `columnflow` does. Now in order to perform some manual operations, we want to convert the `hist` objects to `TH1F/D` objects. If you guys know any hacks to do this, please let us know.
In addition, I was just curious that if it is possible to save the output of the `Plotting` tasks directly to `ROOT` files, separated by the processes maybe. Then, perhaps, one will be able to use some old `ROOT` based scripts for further studies.
Thanks :)

Gourab, Oceane (@oponcet)

**Comment by mafrahm** [comment]
Hi @gsaha009, I am not sure how to convert hist histograms in root objects, but this is already done somewhere in our datacard writer (maybe @riga can point you to an exakt code location). In general for custom processing of histograms, I would not try to use the plotting tasks (their job is to return plots), but implement this as a separate task. There are already some example tasks on how to load histograms from multiple datasets and processes: https://github.com/columnflow/columnflow/blob/master/columnflow/tasks/inspection.py
This baseline structure could be used to build a task that converts histograms into root objects.

**Comment by pkausw** [comment]
Hi @gsaha009 and @oponcet,

Is this still on open question? Regarding manipulating histograms, the UHH HH->bbtautau team implemented a histogram hook that calculates a shape following the ABCD procedure. It's still work in progress, but you can find the code [here](https://github.com/uhh-cms/hh2bbtautau/blob/ana_ABCD/hbt/config/hist_hooks.py). In the current version of columnflow, the plotting tasks have a parameter `hist-hook` which you can use to call this hook. In the function itself, you'll have access to the processes you have requested and all categories, so this should do it. 

Regarding your actual question: as @mafrahm already explained, there is a conversion implemented in the cms inference modules. The actual conversion is done with `uproot`, which can use the hist objects themselves. See [here](https://github.com/columnflow/columnflow/blob/master/columnflow/inference/cms/datacard.py#L361-L391) for an example.

**Comment by gsaha009** [comment]
Thanks @pkausw and @mafrahm for the informative discussion. 
I can close it for the time being.
